core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -579,59 +579,58 @@ private void compactSingleFile(CompactionRequest compactionRequest,
                      break;
                   }
                   log.tracef("No index found for key %s, dropping - assuming lost due to segments removed", key);
-                  scheduledOffset += header.totalLength();
-                  continue;
```

Comment:
The diff may look complex, but all it is is just removing these 2 lines above and then shifting the rest inside an else block. This allows the next if statement to run which removes the index for the value.